zipline/utils/events.py

```diff
@@ -539,7 +539,7 @@ def get_first_trading_day_of_month(self, dt):
         self.month = dt.month
 
         dt = dt.replace(day=1)
-        first_day = (dt if self.cal.is_open_on_day(dt)
+        first_day = (normalize_date(dt) if self.cal.is_open_on_day(dt)
```

Comment:
(that method that @ssanderson described is exactly what I'm in the middle of adding to `ExchangeCalendar` right now)